zipline/pipeline/data/dataset.py

```diff
@@ -115,14 +118,15 @@ class BoundColumn(LoadableTerm):
     inputs = ()
     window_safe = True
 
-    def __new__(cls, dtype, missing_value, dataset, name):
+    def __new__(cls, dtype, missing_value, dataset, name, ndim):
```

Comment:
how do you feel about having boundcolumn have an `ndim` property that returns the ndim of `self.dataset`? This would ensure it is always the same.